addons/sale_product_configurator/views/sale_order_views.xml

```diff
@@ -8,6 +8,7 @@
         <field name="arch" type="xml">
             <xpath expr="//tree/field[@name='product_template_id']" position="attributes">
                 <attribute name="column_invisible">0</attribute>
+                <attribute name="widget">sale_product_field</attribute>
```

Comment:
We could not do that previously, as with the previous modifications the configurator was not installed by default when sale was installed. It will indeed work now.

Reply:
You're maybe confusing the 'configurator', aka the dialog opened (with the logic previously in `product_configurator`), and the widget (ex `sale_product_many2one`, whose logic was defined in `sale` and extended by `event_booth_sale`, `event_sale`, `sale_renting`, `sale_product_configurator` which was even further enhanced by `sale_product_matrix`).

The new logic should revolve around:
`product_many2one` widget (product) : full of hooks but does nothing until a module extends it
`sol_product_many2one` widget (sale, depends on product) : more hooks, still does nothing until a module extends it
`ProductConfiguratorDialog` (product_configurator)
`SaleProductConfiguratorDialog` (sale_product_configurator)

`event_sale`, `event_booth_sale`, `sale_renting` should not change at all js-side, they will keep hooking themselves on `sol_product_many2one`.
`sale_product_configurator` still hooks itself on `sol_product_many2one` and opens his extended version of the configurator (`SaleProductConfiguratorDialog`)
`sale_product_matrix` won't change, and will keep extending the logic in `sale_product_configurator`.

`sale_product_field` (could stay `sol_product_many2one`) should be there even if `sale_product_configurator` is not installed, the same way it was done previously.